hass_nabucasa/files.py

```diff
@@ -111,6 +112,20 @@ async def upload(
             )
 
             self._do_log_response(response)
+            if response.status == 400:
+                # We can try to get some context.
+                error = await response.text()
+                if error and "<Message>" in error and "</Message>" in error:
+                    with contextlib.suppress(AttributeError, IndexError):
+                        # This is ugly but it's the best we can do, we have no control
+                        # over the error message structure, so we try what we can.
+                        parsed = (
+                            error.split("<Message>")[1]
+                            .split("</Message>")[0]
+                            .replace("\n", " ")
+                        )
+                        raise FilesError(f"Failed to upload: (400) {parsed}")
```

Comment:
Instead of the checks, we can use `str.partition` which doesn't fail:
`parsed = error.partition("<Message>")[2].partition("</Message>")[0].replace("\n", " ")`

Reply:
The reason i added checks is incase we only have one of them. We cont want a 6 page exception message

Reply:
Hmm, you mean you don't want the third case?
```
>>> "<Message>Pretty error\nWith a linebreak</Message>".partition("<Message>")[2].partition("</Message>")[0]
'Pretty error\nWith a linebreak'
>>> "Unknown error structure".partition("<Message>")[2].partition("</Message>")[0]
''
>>> "<Message>What is this?".partition("<Message>")[2].partition("</Message>")[0]
'What is this?'
```

Reply:
Yes, as that can be long (last part cut off from response due to pagination or something)

Reply:
That's a valid concern. We *could* fall back to a limited size slice of the text, otherwise let's just merge this PR.